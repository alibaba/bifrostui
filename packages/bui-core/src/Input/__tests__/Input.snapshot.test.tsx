import { snapshotTest } from 'testing';

describe('Input snapshot', () => {
  snapshotTest('Input');
});
