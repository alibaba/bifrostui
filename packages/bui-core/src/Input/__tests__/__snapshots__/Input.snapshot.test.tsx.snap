// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshot Input demo snapshot 0 1`] = `
<div
  className="bui-input"
>
  <input
    className="bui-input-input"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onInput={[Function]}
    placeholder="请填写内容"
    type="text"
    value=""
  />
</div>
`;

exports[`Input snapshot Input demo snapshot 0 2`] = `
<div
  className="bui-input"
>
  <div
    className="bui-input-icon bui-input-icon-start"
  >
    <svg
      className="bui-svg-icon"
      color="#959aa5"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M46.3 8c21.2 0 38.3 17.1 38.3 38.3 0 9.7-3.6 18.5-9.5 25.2l12.1 12.1c1 1 1 2.6 0 3.6s-2.6 1-3.6 0l-12-12.1C64.8 81 56 84.6 46.3 84.6 25.1 84.6 8 67.5 8 46.3S25.1 8 46.3 8zm0 6C28.5 14 14 28.5 14 46.3s14.4 32.3 32.3 32.3 32.3-14.4 32.3-32.3S64.1 14 46.3 14z\\"/>",
        }
      }
      focusable="false"
      viewBox="0 0 96 96"
    />
  </div>
  <input
    className="bui-input-input"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onInput={[Function]}
    placeholder="请填写内容"
    type="text"
    value=""
  />
</div>
`;

exports[`Input snapshot Input demo snapshot 0 3`] = `
<div
  className="bui-input"
>
  <input
    className="bui-input-input"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onInput={[Function]}
    placeholder="请填写内容"
    type="text"
    value=""
  />
  <div
    className="bui-input-icon bui-input-icon-end"
  >
    <svg
      className="bui-svg-icon"
      color="#959aa5"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M46.3 8c21.2 0 38.3 17.1 38.3 38.3 0 9.7-3.6 18.5-9.5 25.2l12.1 12.1c1 1 1 2.6 0 3.6s-2.6 1-3.6 0l-12-12.1C64.8 81 56 84.6 46.3 84.6 25.1 84.6 8 67.5 8 46.3S25.1 8 46.3 8zm0 6C28.5 14 14 28.5 14 46.3s14.4 32.3 32.3 32.3 32.3-14.4 32.3-32.3S64.1 14 46.3 14z\\"/>",
        }
      }
      focusable="false"
      viewBox="0 0 96 96"
    />
  </div>
</div>
`;

exports[`Input snapshot Input demo snapshot 0 4`] = `
Array [
  <div
    className="bui-input"
  >
    <input
      className="bui-input-input"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onInput={[Function]}
      placeholder="请填写内容"
      type="text"
      value=""
    />
  </div>,
  <div
    style={
      Object {
        "margin": "20px 0 10px",
      }
    }
  >
    非受控也会清空输入框：
  </div>,
  <div
    className="bui-input"
  >
    <input
      className="bui-input-input"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onInput={[Function]}
      type="text"
      value="会清空内容"
    />
  </div>,
]
`;

exports[`Input snapshot Input demo snapshot 0 5`] = `
<div
  className="bui-input bui-input-disabled"
>
  <input
    className="bui-input-input"
    disabled={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onInput={[Function]}
    placeholder="请填写内容"
    type="text"
    value=""
  />
</div>
`;

exports[`Input snapshot Input demo snapshot 0 6`] = `
<div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "marginBottom": "40px",
      }
    }
  >
    受控组件：
    <button
      className="bui-btn bui-btn-outlined bui-btn-medium bui-btn-default"
      onClick={[Function]}
      type="button"
    >
      数量加10
    </button>
    <br />
    <div
      className="bui-input"
    >
      <input
        className="bui-input-input"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        type="text"
        value="0"
      />
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    非受控组件：
    <br />
    <button
      className="bui-btn bui-btn-outlined bui-btn-medium bui-btn-default"
      onClick={[Function]}
      type="button"
    >
      数量减10(操作defaultValue无效)
    </button>
    <br />
    <button
      className="bui-btn bui-btn-outlined bui-btn-medium bui-btn-default"
      onClick={[Function]}
      type="button"
    >
      数量乘2(通过DOM操作value有效)
    </button>
    <br />
    <div
      className="bui-input"
    >
      <input
        className="bui-input-input"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        type="text"
        value="100"
      />
    </div>
  </div>
</div>
`;

exports[`Input snapshot Input demo snapshot 0 7`] = `
<div
  className="bui-input"
>
  <input
    className="bui-input-input"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onInput={[Function]}
    placeholder="number类型"
    type="number"
    value=""
  />
</div>
`;
